### PR TITLE
Sort the time to display the earliest and return number of slots

### PR DESCRIPTION
**Pull request by aroltsch**
connects #3275 

image

### CODE REVIEW

File: app/repositories/hearing_repository.rb [hunk 1]
```diff
@@ -70,5 +76,21 @@ def hearings_for(case_hearings)
       end
     end
     # :nocov:
+
+    def vacols_attributes(vacols_record)
+      attrs = {
+        vacols_record: vacols_record,
+        venue_key: vacols_record.hearing_venue,
+        disposition: VACOLS::CaseHearing::HEARING_DISPOSITIONS[vacols_record.hearing_disp.try(:to_sym)],
+        date: vacols_record.hearing_date,
```
Comment (aroltsch) [inline app/repositories/hearing_repository.rb]: We will send the hearing date the way it is and FE will determine the timezone based on the venue. According to Jed, the ROs schedule the veterans on the Travel Boards so the times reflect the time zone of the local RO.

File: app/models/vacols/case_hearing.rb [hunk 2]
```diff
@@ -72,23 +72,16 @@ def select_hearings
...
+             :hearing_disp, :hearing_pkseq,
+             :hearing_date, :hearing_type,
+             :notes1, :folder_nr,
+             :vdkey, :aod,
+             :holddays, :tranreq,
+             :repname, :addon,
+             :board_member, :mduser,
+             :mdtime, :sattyid)
         .joins("left outer join vacols.staff on staff.sattyid = board_member")
+        .includes(:brieff)
```
Comment (aroltsch) [inline app/models/vacols/case_hearing.rb]: This will speed things up a bit.

Comment (sharonwarner) [reply app/models/vacols/case_hearing.rb]: Nice. I was on UAT a bunch yesterday, and I was surprised by how slowly this page loads.

File: app/repositories/hearing_repository.rb [hunk 3]
```diff
@@ -22,29 +22,35 @@ def load_vacols_data(hearing)
...
       hearing
     end
 
+    # STAFF.STC2 is the Travel Board limit for Mon and Fri
+    # STAFF.STC3 is the Travel Board limit for Tue, Wed, Thur
+    # STAFF.STC4 is the Video limit
+    def slots_based_on_type(staff:, type:, date:)
+      case type
+      when :central_office
+        11
```
Comment (sharonwarner) [inline app/repositories/hearing_repository.rb]: Haha. I love how easy the CO number is.

File: app/repositories/hearing_repository.rb [hunk 4]
```diff
@@ -70,5 +76,21 @@ def hearings_for(case_hearings)
...
+        date: vacols_record.hearing_date,
+        representative_name: vacols_record.repname,
+        aod: VACOLS::CaseHearing::HEARING_AODS[vacols_record.aod.try(:to_sym)],
+        hold_open: vacols_record.holddays,
+        transcript_requested: VACOLS::CaseHearing::BOOLEAN_MAP[vacols_record.tranreq.try(:to_sym)],
+        add_on: VACOLS::CaseHearing::BOOLEAN_MAP[vacols_record.addon.try(:to_sym)],
+        notes: vacols_record.notes1,
+        master_record: vacols_record.master_record?
+      }.merge(values_based_on_type(vacols_record))
+    end
```
Comment (sharonwarner) [inline app/repositories/hearing_repository.rb]: I love getting this out of the :nocov: section!

File: app/models/vacols/case_hearing.rb (hunk 2 above)
Comment (aroltsch) [reply app/models/vacols/case_hearing.rb]: Yeah, I think we can make more improvements. The problem is on the appeal. Each hearing load causes 3 selects:
```
VACOLS::Case Load (26.0ms)  SELECT  "VACOLS"."BRIEFF".* FROM "VACOLS"."BRIEFF" WHERE "VACOLS"."BRIEFF"."BFKEY" = :a1 AND ROWNUM <= 1   [["bfkey", "2372015"]] VACOLS::Folder Load (26.7ms)  SELECT "VACOLS"."FOLDER".* FROM "VACOLS"."FOLDER" WHERE "VACOLS"."FOLDER"."TICKNUM" IN ('2372015')
VACOLS::Correspondent Load (48.9ms)  SELECT "VACOLS"."CORRES".* FROM "VACOLS"."CORRES" WHERE "VACOLS"."CORRES"."STAFKEY" IN ('299384')
```

File: lib/fakes/hearing_repository.rb [hunk 5]
```diff
@@ -36,6 +36,10 @@ def self.load_vacols_data(hearing)
     true
   end
 
+  def self.number_of_slots(*)
```
Comment (kierachell) [inline lib/fakes/hearing_repository.rb]: Sometimes the STC2-4 is nil. What happens then?

**Pull request merged**